realm/src/main/java/io/realm/Realm.java

```diff
@@ -154,6 +155,9 @@
     private static final Map<String, AtomicInteger> globalOpenInstanceCounter =
             new ConcurrentHashMap<String, AtomicInteger>();
 
+    // Thread Pool for all async operations (Query & Write transaction)
+    public static final ExecutorService asyncQueryExecutor = Executors.newFixedThreadPool(Runtime.getRuntime().availableProcessors() * 2 + 1);
```

Comment:
I know AsyncTask use the same ThreadPool size policy, but are we aware of the tradeoffs involved? Ie. A fixed size ThreadPool will actually block until there is new threads available, so in, albeit very weird edge cases, you might actually block certain parts of the program from receiving query results?